### DISCUSSION TITLE
Script Help

### DISCUSSION THREAD

**Discussion by nikoniko78** [nikoniko78/scripts-java-physion#7] [Announcements]
There has been a script I want to use in one of my scenes. It is found in the reply to [Help Please 2](https://app.physion.net/scenes/help-please-2) by Sirfryingpan. Please reply if you find this scene.

**Comment by Carstyle476** [comment]
It's probably the DoorController script from Sirfryingpan's coin game.

**Comment by nikoniko78** [comment]
Since you created this script, could you please educate me on how to use it? I have no specification on which type to put it on. Do i put it on the motor, or the sensor?

**Comment by Carstyle476** [comment]
### WALL OF TEXT INCOMING

**Step 1. Setup**
- Place the script in the trigger node - the BodyNode that will trigger the door's motor to open.
- Add 3 values to the trigger node's User Data.
 - The motor ID, name it "motorId"
 - The motor speed, name it "speed"
 - The amount of time to open the door, name it "openTime"

**Step 2. Usage**
- Test the script by making another BodyNode collide with the trigger node.
 - If the door goes the wrong way/doesn't open, flip the sign of the speed value.
 - If the door doesn't move at all or the wrong motor moves, check the motor ID value.
 - If the door isn't open for long enough, increase the open time value.
- Optionally make the trigger node a sensor instead.

Finish!

**Reply by nikoniko78** [reply]
Thanks!
